common/OpTestSSH.py

```diff
@@ -155,52 +159,131 @@ def get_console(self):
+            combo_io = console.before + console.after
+            retry_list_output += combo_io.splitlines()
+            matching = [xs for xs in sudo_responses if any(xs in xa for xa in console.after.splitlines())]
+            if len(matching):
+              echo_rc = 1
+              rc = -1 # use to flag the failure next
+          if rc == 0:
+            retry_list_output += self.set_env(console)
+            echo_rc = 0
+            break
+          elif a == 2:
+            echo_rc = 1
+            break
+          elif (rc == 2) or (rc == 3):
+            raise CommandFailed(command, 'Retry Password TIMEOUT ' + ''.join(retry_list_output), -1)
+        return retry_list_output, echo_rc
 
+    def handle_password(self, console, command):
+        # this is for run_command 'sudo -s' or the like
+        handle_list_output = []
+        failure_list_output = []
+        pre_combo_io = console.before + console.after
+        console.sendline(self.password)
+        rc = console.expect([".*#", "try again.", pexpect.TIMEOUT, pexpect.EOF])
+        if (rc == 0) or (rc == 1):
+          combo_io = pre_combo_io + console.before + console.after
+          handle_list_output += combo_io.splitlines()
+          matching = [xs for xs in sudo_responses if any(xs in xa for xa in console.after.splitlines())]
+          if len(matching):
+            # remove the expect prompt since matched generic #
+            del handle_list_output[-1]
+            echo_rc = 1
+            rc = -1 # use this to flag the failure next
+        if rc == 0:
+          # with unknown prompts and unknown environment unable to capture echo $?
+          echo_rc = 0
+          self.set_env(console)
+          list_output = handle_list_output
+        elif rc == 1:
+          retry_list_output, echo_rc = self.retry_password(console, command)
+          list_output = (handle_list_output + retry_list_output)
+        else:
+          if (rc == 2) or (rc == 3):
+            failure_list_output += ['Password Problem/TIMEOUT ']
+            failure_list_output += pre_combo_io.splitlines()
+          # timeout path needs access to output
+          # handle_list_output empty if timeout or EOF
+          failure_list_output += handle_list_output
+          raise CommandFailed(command, ''.join(failure_list_output), -1)
+        return list_output, echo_rc
+
+    def run_command(self, command, timeout=60):
+        running_sudo_s = False
+        extra_sudo_output = False
+        console = self.get_console()
+        expect_prompt = self.build_prompt() + "$"
+        console.sendline(command)
+        console.expect("\n") # removes the echo of command from output
+        if command == 'sudo -s':
+          running_sudo_s = True
+          # special case to catch loss of env
+          rc = console.expect([".*#", r"[Pp]assword", pexpect.TIMEOUT, pexpect.EOF])
+        else:
+          rc = console.expect([expect_prompt, r"[Pp]assword", pexpect.TIMEOUT, pexpect.EOF])
```

Comment:
And also run_command should provide flexible timeout for the user to consider. Please add timeout for expect functions everywhere. 
```
 +          rc = console.expect([expect_prompt, r"[Pp]assword", pexpect.TIMEOUT, pexpect.EOF], timeout=timeout) 
```
Otherwise it timeouts for default 30 seconds provided by OPexpect.